powerline/segments/common/bat.py

```diff
@@ -179,32 +184,52 @@ def battery(pl, format='{capacity:3.0%}', steps=5, gamify=False, full_heart='O',
 		Number of discrete steps to show between 0% and 100% capacity if gamify
 		is True.
 	:param bool gamify:
-		Measure in hearts (♥) instead of percentages. For full hearts 
-		``battery_full`` highlighting group is preferred, for empty hearts there 
+		Measure in hearts (♥) instead of percentages. For full hearts
+		``battery_full`` highlighting group is preferred, for empty hearts there
 		is ``battery_empty``.
 	:param str full_heart:
 		Heart displayed for “full” part of battery.
 	:param str empty_heart:
 		Heart displayed for “used” part of battery. It is also displayed using
-		another gradient level and highlighting group, so it is OK for it to be 
-		the same as full_heart as long as necessary highlighting groups are 
+		another gradient level and highlighting group, so it is OK for it to be
+		the same as full_heart as long as necessary highlighting groups are
 		defined.
+	:param str online:
+		If computer is connected to a power supply this symbol is prepended to the segment.
+	:param str offline:
+		If computer is NOT connected to a power supply this symbol is prepended to the segment.
 
-	``battery_gradient`` and ``battery`` groups are used in any case, first is 
+	``battery_gradient`` and ``battery`` groups are used in any case, first is
 	preferred.
 
-	Highlight groups used: ``battery_full`` or ``battery_gradient`` (gradient) or ``battery``, ``battery_empty`` or ``battery_gradient`` (gradient) or ``battery``.
+	Highlight groups used: ``battery_full`` or ``battery_gradient`` (gradient) or ``battery``,
+	``battery_empty`` or ``battery_gradient`` (gradient) or ``battery``,
+	``battery_online`` or ``battery_ac_state`` or ``battery_gradient`` (gradient) or ``battery``,
+	``battery_offline`` or ``battery_ac_state`` or ``battery_gradient`` (gradient) or ``battery``.
 	'''
 	try:
-		capacity = _get_capacity(pl)
+		capacity, ac_powered = _get_battery_status(pl)
 	except NotImplementedError:
-		pl.info('Unable to get battery capacity.')
+		pl.info('Unable to get battery status.')
 		return None
+
 	ret = []
 	if gamify:
 		denom = int(steps)
 		numer = int(denom * capacity / 100)
 		ret.append({
+			'contents': online if ac_powered else '',
+			'draw_inner_divider': False,
+			'highlight_groups': ['battery_online', 'battery_ac_state', 'battery_gradient', 'battery'],
+			'gradient_level': 0,
+		})
+		ret.append({
+			'contents': '' if ac_powered else offline,
+			'draw_inner_divider': False,
+			'highlight_groups': ['battery_offline', 'battery_ac_state', 'battery_gradient', 'battery'],
+			'gradient_level': 0,
+		})
```

Comment:
It is bad idea to have segment with empty contents. You just need to have `'contents': online if ac_powered else offline` and below `'highlight_groups' : ['battery_online' if ac_powered else 'battery_online', …`.